backend/grants/admin.py

```diff
@@ -303,24 +291,47 @@ def create_grant_vouchers_on_pretix(modeladmin, request, queryset):
             )
             continue
 
-        voucher_code = _generate_voucher_code("GRANT")
-        pretix_voucher = create_voucher(
-            conference=grant.conference,
-            code=voucher_code,
-            comment=f"Voucher for user_id={grant.user_id}",
-            tag="grants",
-            quota_id=grant.conference.pretix_conference_voucher_quota_id,
-            price_mode="set",
-            value="0.00",
-        )
+        existing_voucher = existing_vouchers_by_user_id.get(grant.user_id)
 
-        pretix_voucher_id = pretix_voucher["id"]
-        grant.pretix_voucher_id = pretix_voucher_id
-        grant.voucher_code = voucher_code
-        grant.save()
-        count += 1
+        if not existing_voucher:
+            create_addition_admin_log_entry(
+                request.user,
+                grant,
+                change_message="Created voucher for this grant",
+            )
+
+            vouchers_to_create.append(
+                ConferenceVoucher(
+                    conference_id=grant.conference_id,
+                    user_id=grant.user_id,
+                    voucher_code=ConferenceVoucher.generate_code(),
+                    voucher_type=ConferenceVoucher.VoucherType.GRANT,
+                )
+            )
+            continue
+
+        if existing_voucher.voucher_type == ConferenceVoucher.VoucherType.CO_SPEAKER:
```

Comment:
What if the type is `SPEAKER`??

Reply:
If the type is speaker we already give them a 100% discount and I think it is more correct to keep their voucher marked as SPEAKER, as technically their ticket is already included in the speakers budget

Reply:
oh I see, right!